internal/olm/operator/internal/registry_pod_test.go

```diff
@@ -0,0 +1,139 @@
+			})
+
+			It("should create the RegistryPod successfully", func() {
+				Expect(err).To(BeNil())
+				Expect(rp).NotTo(BeNil())
+				Expect(rp.pod.Name).To(Equal(expectedPodName))
+				Expect(rp.pod.Namespace).To(Equal(rp.Namespace))
+				Expect(rp.pod.Spec.Containers[0].Name).To(Equal(defaultContainerName))
+				if len(rp.pod.Spec.Containers) > 0 {
+					if len(rp.pod.Spec.Containers[0].Ports) > 0 {
+						Expect(rp.pod.Spec.Containers[0].Ports[0].ContainerPort).To(Equal(rp.GRPCPort))
+					}
+				}
+			})
+
+			It("should return a valid container command", func() {
+				output, err := rp.getContainerCmd()
+
+				Expect(err).To(BeNil())
+				Expect(output).Should(Equal(expectedOutput))
+			})
+
+			It("should create registry pod successfully", func() {
+				err := rp.Create(context.Background())
+
+				Expect(err).To(BeNil())
+			})
+
+		})
+
+		Context("with invalid registry pod values", func() {
+
+			It("should error when bundle image is not provided", func() {
+				expectedErr := "bundle image cannot be empty"
+
+				_, err := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"", "default")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should not create a registry pod when namespace is not provided", func() {
+				expectedErr := "namespace cannot be empty"
+
+				_, err := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"quay.io/example/example-operator-bundle:0.2.0", "")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should not create a registry pod when database path is not provided", func() {
+				expectedErr := "registry database path cannot be empty"
+
+				_, err := NewRegistryPod(newFakeClient(), "",
+					"quay.io/example/example-operator-bundle:0.2.0", "default")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should error when invalid bundle image is provided", func() {
+				expectedErr := "invalid bundle image name"
+
+				_, err := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"non/existent/image", "default")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should not accept any other bundle add mode other than semver or replaces", func() {
+				expectedErr := "invalid bundle mode"
+
+				rp, _ := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"quay.io/example/example-operator-bundle:0.2.0", "default")
+				rp.BundleAddMode = "invalid"
+
+				err := rp.validate()
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			// todo(rashmigottipati): add test to check Create returning error
+			// todo(rashmigottipati): add test to check VerifyPodRunning returning error
```

Comment:
PS. Should not it add now? Has some reason for it be a follow-up?

Reply:
Went down that path and it seems more tricky than anticipated, due to the fake client. 
IMO, it's not blocking the PR merge, so prefer to do it in a follow-up.